crates/invoker-impl/src/state_machine_manager.rs

```diff
@@ -73,15 +73,57 @@ where
     }
 
     #[inline]
-    pub(super) fn remove_invocation(
+    pub(super) fn handle_for_invocation<R>(
         &mut self,
         partition: PartitionLeaderEpoch,
         invocation_id: &InvocationId,
-    ) -> Option<(&mpsc::Sender<Effect>, &IR, InvocationStateMachine)> {
+        invocation_epoch: InvocationEpoch,
+        f: impl FnOnce(&mpsc::Sender<Effect>, &mut InvocationStateMachine) -> R,
+    ) -> Option<R> {
+        if let Some((tx, ism)) =
+            self.resolve_invocation_with_epoch(partition, invocation_id, invocation_epoch)
+        {
+            Some(f(tx, ism))
+        } else {
+            // If no state machine
+            trace!("No state machine found for selected server header");
```

Comment:
What does server header relates to?

Reply:
it's the header that tells us which SDK the user is using.